File: SpaceSalvagers/Resources/GenerateLaunchImages.swift
import UIKit
import CoreGraphics

class LaunchImageGenerator {
    
    static func generateLaunchBackground(size: CGSize) -> UIImage {
        let renderer = UIGraphicsImageRenderer(size: size)
        
        return renderer.image { context in
            let rect = CGRect(origin: .zero, size: size)
            
            // Dark space background
            UIColor(red: 0.02, green: 0.02, blue: 0.08, alpha: 1.0).setFill()
            context.fill(rect)
            
            // Add stars
            for _ in 0..<100 {
                let x = CGFloat.random(in: 0...size.width)
                let y = CGFloat.random(in: 0...size.height)
                let starSize = CGFloat.random(in: 0.5...2.0)
                let alpha = CGFloat.random(in: 0.3...1.0)
                
                UIColor.white.withAlphaComponent(alpha).setFill()
                context.cgContext.fillEllipse(in: CGRect(x: x - starSize/2, y: y - starSize/2, width: starSize, height: starSize))
            }
            
            // Add nebula gradient
            let gradient = CGGradient(
                colorsSpace: CGColorSpaceCreateDeviceRGB(),
                colors: [
                    UIColor(red: 0.1, green: 0.0, blue: 0.3, alpha: 0.2).cgColor,
                    UIColor(red: 0.0, green: 0.1, blue: 0.4, alpha: 0.1).cgColor,
                    UIColor.clear.cgColor
                ] as CFArray,
                locations: [0, 0.5, 1]
            )!
            
            context.cgContext.drawRadialGradient(
                gradient,
                startCenter: CGPoint(x: size.width * 0.3, y: size.height * 0.3),
                startRadius: 0,
                endCenter: CGPoint(x: size.width * 0.3, y: size.height * 0.3),
                endRadius: size.width * 0.5,
                options: []
            )
        }
    }
    
    static func generateLaunchLogo(size: CGSize) -> UIImage {
        let renderer = UIGraphicsImageRenderer(size: size)
        
        return renderer.image { context in
            let rect = CGRect(origin: .zero, size: size)
            
            // Clear background
            UIColor.clear.setFill()
            context.fill(rect)
            
            // Draw hexagonal station core
            let centerX = size.width / 2
            let centerY = size.height / 2
            let radius = min(size.width, size.height) * 0.35
            
            // Outer glow
            for i in stride(from: 3, through: 0, by: -1) {
                let glowRadius = radius + CGFloat(i) * 10
                let alpha = 0.2 / CGFloat(i + 1)
                
                drawHexagon(
                    context: context.cgContext,
                    center: CGPoint(x: centerX, y: centerY),
                    radius: glowRadius,
                    fillColor: UIColor(red: 0, green: 0.8, blue: 1.0, alpha: alpha),
                    strokeColor: UIColor.clear
                )
            }
            
            // Main hexagon
            drawHexagon(
                context: context.cgContext,
                center: CGPoint(x: centerX, y: centerY),
                radius: radius,
                fillColor: UIColor(red: 0.05, green: 0.15, blue: 0.25, alpha: 1.0),
                strokeColor: UIColor(red: 0, green: 0.8, blue: 1.0, alpha: 1.0)
            )
            
            // Inner core
            let innerRadius = radius * 0.5
            drawHexagon(
                context: context.cgContext,
                center: CGPoint(x: centerX, y: centerY),
                radius: innerRadius,
                fillColor: UIColor(red: 0, green: 0.6, blue: 0.8, alpha: 0.8),
                strokeColor: UIColor(red: 0, green: 1.0, blue: 1.0, alpha: 1.0)
            )
            
            // Center glow
            let glowGradient = CGGradient(
                colorsSpace: CGColorSpaceCreateDeviceRGB(),
                colors: [
                    UIColor(red: 0, green: 1.0, blue: 1.0, alpha: 0.8).cgColor,
                    UIColor(red: 0, green: 0.8, blue: 1.0, alpha: 0.3).cgColor,
                    UIColor.clear.cgColor
                ] as CFArray,
                locations: [0, 0.5, 1]
            )!
            
            context.cgContext.drawRadialGradient(
                glowGradient,
                startCenter: CGPoint(x: centerX, y: centerY),
                startRadius: 0,
                endCenter: CGPoint(x: centerX, y: centerY),
                endRadius: innerRadius,
                options: []
            )
        }
    }
    
    private static func drawHexagon(context: CGContext, center: CGPoint, radius: CGFloat, fillColor: UIColor, strokeColor: UIColor) {
        let path = UIBezierPath()
        
        for i in 0..<6 {
            let angle = CGFloat(i) * CGFloat.pi / 3.0 - CGFloat.pi / 6.0
            let x = center.x + radius * cos(angle)
            let y = center.y + radius * sin(angle)
            
            if i == 0 {
                path.move(to: CGPoint(x: x, y: y))
            } else {
                path.addLine(to: CGPoint(x: x, y: y))
            }
        }
        
        path.close()
        
        fillColor.setFill()
        path.fill()
        
        strokeColor.setStroke()
        path.lineWidth = 2
        path.stroke()
    }
    
    static func saveImages() {
        // Generate and save background images
        let bgSizes: [(CGSize, String)] = [
            (CGSize(width: 320, height: 568), "launch-bg@1x.png"),
            (CGSize(width: 750, height: 1334), "launch-bg@2x.png"),
            (CGSize(width: 1242, height: 2208), "launch-bg@3x.png")
        ]
        
        for (size, filename) in bgSizes {
            let image = generateLaunchBackground(size: size)
            if let data = image.pngData() {
                let path = "/Users/wlprinsloo/Documents/Projects/Tower-game/SpaceSalvagers/Resources/Assets.xcassets/LaunchBackground.imageset/\(filename)"
                try? data.write(to: URL(fileURLWithPath: path))
                print("Saved: \(path)")
            }
        }
        
        // Generate and save logo images
        let logoSizes: [(CGSize, String)] = [
            (CGSize(width: 200, height: 200), "launch-logo@1x.png"),
            (CGSize(width: 400, height: 400), "launch-logo@2x.png"),
            (CGSize(width: 600, height: 600), "launch-logo@3x.png")
        ]
        
        for (size, filename) in logoSizes {
            let image = generateLaunchLogo(size: size)
            if let data = image.pngData() {
                let path = "/Users/wlprinsloo/Documents/Projects/Tower-game/SpaceSalvagers/Resources/Assets.xcassets/LaunchLogo.imageset/\(filename)"
                try? data.write(to: URL(fileURLWithPath: path))
                print("Saved: \(path)")
            }
        }
    }
}

// Generate images when run
LaunchImageGenerator.saveImages()

File: generate_launch_images.py
#!/usr/bin/env python3

from PIL import Image, ImageDraw
import os

def create_launch_background(width, height):
    """Create a space-themed background"""
    img = Image.new('RGB', (width, height), color=(5, 5, 20))
    draw = ImageDraw.Draw(img)
    
    # Add some stars
    import random
    for _ in range(100):
        x = random.randint(0, width)
        y = random.randint(0, height)
        brightness = random.randint(100, 255)
        draw.point((x, y), fill=(brightness, brightness, brightness))
    
    return img

def create_launch_logo(size):
    """Create a hexagonal station core logo"""
    img = Image.new('RGBA', (size, size), (0, 0, 0, 0))
    draw = ImageDraw.Draw(img)
    
    # Draw a hexagon
    center_x = size // 2
    center_y = size // 2
    radius = size // 3
    
    # Calculate hexagon points
    import math
    points = []
    for i in range(6):
        angle = math.pi / 3 * i - math.pi / 6
        x = center_x + radius * math.cos(angle)
        y = center_y + radius * math.sin(angle)
        points.append((x, y))
    
    # Draw filled hexagon with cyan color
    draw.polygon(points, fill=(0, 204, 255, 200), outline=(0, 255, 255, 255))
    
    # Draw inner hexagon
    inner_radius = radius // 2
    inner_points = []
    for i in range(6):
        angle = math.pi / 3 * i - math.pi / 6
        x = center_x + inner_radius * math.cos(angle)
        y = center_y + inner_radius * math.sin(angle)
        inner_points.append((x, y))
    
    draw.polygon(inner_points, fill=(0, 255, 255, 255))
    
    return img

# Create directories if they don't exist
base_path = "/Users/wlprinsloo/Documents/Projects/Tower-game/SpaceSalvagers/Resources/Assets.xcassets"
os.makedirs(f"{base_path}/LaunchBackground.imageset", exist_ok=True)
os.makedirs(f"{base_path}/LaunchLogo.imageset", exist_ok=True)

# Generate background images
bg_sizes = [
    (320, 568, "launch-bg@1x.png"),
    (750, 1334, "launch-bg@2x.png"),
    (1242, 2208, "launch-bg@3x.png")
]

for width, height, filename in bg_sizes:
    img = create_launch_background(width, height)
    img.save(f"{base_path}/LaunchBackground.imageset/{filename}")
    print(f"Created {filename}")

# Generate logo images
logo_sizes = [
    (200, "launch-logo@1x.png"),
    (400, "launch-logo@2x.png"),
    (600, "launch-logo@3x.png")
]

for size, filename in logo_sizes:
    img = create_launch_logo(size)
    img.save(f"{base_path}/LaunchLogo.imageset/{filename}")
    print(f"Created {filename}")

print("All launch images created successfully!")